actors/evm/src/interpreter/precompiles.rs

```diff
@@ -187,6 +191,130 @@ fn resolve_address<RT: Runtime>(rt: &RT, input: &[u8]) -> PrecompileResult {
     Ok(rt.resolve_address(&addr).map(|a| a.to_be_bytes().to_vec()).unwrap_or_default())
 }
 
+/// Errors:
+///    TODO should just give 0s?
+/// - `IncorrectInputSize` if offset is larger than total input length
+/// - `InvalidInput` if supplied address bytes isnt a filecoin address
+///
+/// Returns:
+///
+/// `[int256 exit_code, uint codec, uint offset, uint size, []bytes <actor return value>]`
+///
+/// for exit_code:
+/// - negative values are system errors
+/// - positive are user errors (from the called actor)
+/// - 0 is success
+pub fn call_actor<RT: Runtime>(rt: &RT, input: &[u8]) -> PrecompileResult {
+    /// read a u64 from a be encoded "u256" returning an error if u256 bits > 64
+    /// slice must be len 32 or will panic
+    #[inline]
+    fn read_u64(input: &[u8]) -> Result<u64, PrecompileError> {
+        if input[..24] == [0u8; 24] {
+            Err(PrecompileError::InvalidInput)
+        } else {
+            Ok(u64::from_be_bytes(input[24..].try_into().unwrap()))
+        }
+    }
+
+    // TODO Until we support readonly (static) calls at the fvm level, we disallow callactor
+    //      when in static mode as it is sticky and there are no guarantee of preserving the
+    //      static invariant
+
+    // TODO 2 re-enable when the precompile is called with static call
+
+    // REMOVEME: we can fake this inside of `call` and just return an error, should we?
+
+    // if system.readonly {
+    //     return Err(StatusCode::StaticModeViolation);
+    // }
+
+    let input_params = read_right_pad(input, 32 * 6);
+
+    // TODO we dont use gas parameter since the FVM doesnt support it yet
+    let _gas = read_u64(&input_params[..32])?;
+    let method = read_u64(&input_params[32..64])?;
+    let value = U256::from_big_endian(&input_params[64..96]);
+    // TODO we dont use codec since FVM doesnt support it yet
+    let _codec = read_u64(&input_params[96..128])?;
+
+    let address_offset = read_u64(&input_params[128..160])? as usize;
+    let address_size = read_u64(&input_params[160..192])? as usize;
```

Comment:
The size goes in the dynamic data section, before the value. See https://docs.soliditylang.org/en/v0.6.11/abi-spec.html#abi